angular/src/app/services/wallet-manager.ts

```diff
@@ -186,14 +189,58 @@ export class WalletManager {
     return data;
   }
 
+  async selectUtxos(utxos: { txId: string; vout: number; value: number }[], targets: { address: string; value: number }[], feeRate: number): Promise<CoinSelectionResult> {
+    // When user manually edits fee, the value becomes string and not number and the coinselect library is very strict on type.
+    if (typeof String(feeRate)) {
+      feeRate = Number(feeRate);
+    }
+
+    // The coin select library will result in fee rate that is right below what is specified, so for example
+    // 10 can result in fees at around 9.9-9.5 etc. and that won't be accepted by the nodes when broadcasted.
+    feeRate = feeRate + 1;
+
+    const totalInputAmount = utxos.reduce((partialSum, a) => partialSum + a.value, 0);
+    const totalOutputAmount = targets.reduce((partialSum, a) => partialSum + a.value, 0);
+    let result: CoinSelectionResult = null;
+
+    // We could/should also check if the output is within the threshold of dust or not.
+    if (totalInputAmount == totalOutputAmount) {
+      // Run coinsplit instead of coinselect and remove the specified output for now.
```

Comment:
I don't fully understand this comment, I suppose its the underline lib logic?
Why remove an output if the amounts are the same?

Reply:
This is when user want to use the max-send feature, then this coinsplit operation must be run like this to calculate fee correctly.